Mathlib/RingTheory/Flat/Localization.lean

```diff
@@ -30,3 +39,65 @@ theorem IsLocalization.flat : Module.Flat R S :=
     simpa [this, - Subtype.val_injective] using Subtype.val_injective
 
 instance Localization.flat : Module.Flat R (Localization p) := IsLocalization.flat _ p
+
+namespace Module
+
+include p in
+theorem flat_iff_localization : Flat S M ↔ Flat R M :=
+  have := isLocalizedModule_id p M S
+  have := IsLocalization.flat S p
+  ⟨fun _ ↦ .trans R S M, fun _ ↦ .of_isLocalizedModule S p .id⟩
+
+private lemma aux (I : Ideal R) (s : Submonoid R) :
+    have hM := isBaseChange s (Localization s) (mkLinearMap s M)
+    have hIM := isBaseChange s (Localization s) (mkLinearMap s (I ⊗[R] M))
+    let e := (hIM.equiv.restrictScalars R).symm ≪≫ₗ
+      leftComm _ _ _ _ ≪≫ₗ (hM.equiv.restrictScalars R).lTensor I
+    LocalizedModule.map s (TensorProduct.lift <| lsmul R M ∘ₗ I.subtype) =
+      TensorProduct.lift (lsmul R (LocalizedModule s M) ∘ₗ I.subtype) ∘ₗ e.toLinearMap := by
+  refine linearMap_ext s (mkLinearMap s _) (mkLinearMap s _) ?_
+  ext m i
+  show LocalizedModule.map s _ _ = _
+  rw [mkLinearMap_apply]
```

Comment:
This gets rids of the `show`: it's just a bit long to write. The main reason it can't be a single `simp` call is there are two places where `rw [mkLinearMap_apply]` would fire, but we must control it to rewrite precisely one place for the proof to work.
```suggestion
  rw [AlgebraTensorModule.curry_apply, curry_apply, restrictScalars_apply, LinearMap.comp_apply,
    restrictScalars_apply, mkLinearMap_apply]
```

Reply:
Even if its lengthy, I still prefer this over the `show`.